.github/workflows/docs.yml

```
@@ -0,0 +1,43 @@
+          python -m pip install --upgrade pip
+          pip install -r docs/requirements-docs.txt
+          sudo apt-get update
+          sudo apt-get install -y doxygen
+
+      - name: Build documentation
+        run: |
+          ./scripts/build_docs.sh
+
+      - name: Deploy to Jekyll site
+        if: github.ref == 'refs/heads/main'  # Only deploy on main branch
+        uses: peaceiris/actions-gh-pages@v3
```

actionlint (1.7.9)

the runner of "peaceiris/actions-gh-pages@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)